### PR TITLE
LibAudio: Change the API to allow for multiple file formats

### PR DESCRIPTION
**Pull request by metalvoidzz**
This patch adds a generic interface for loading any kind of supported audio file format, instead of just WAV.

### CODE REVIEW

File: Libraries/LibAudio/WavLoader.cpp [hunk 1]
```diff
@@ -33,23 +33,29 @@
...
-RefPtr<Buffer> WavLoader::get_more_samples(size_t max_bytes_to_read_from_input)
+bool WavLoaderPlugin::sniff()
 {
-#ifdef AWAVLOADER_DEBUG
+    return valid;
+}
+
+RefPtr<Buffer> WavLoaderPlugin::get_more_samples(size_t max_bytes_to_read_from_input)
+{
+#ifdef AWavLoaderPlugin_DEBUG
```
Comment (awesomekling) [inline Libraries/LibAudio/WavLoader.cpp:58]: Please don't use lowercase letters in macros. Also, if you change a debug macro, you need to update CMakeLists.txt (see the CI failure.)

File: Libraries/LibAudio/Loader.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <LibAudio/WavLoader.h>
+
+namespace Audio {
+
+Loader::Loader(const StringView& path)
```
Comment (awesomekling) [inline Libraries/LibAudio/Loader.cpp:31]: At some point we should also support input not coming from a file on disk. (To support streaming etc.)

File: Libraries/LibAudio/WavLoader.cpp (hunk 1 above)
Comment (metalvoidzz) [reply Libraries/LibAudio/WavLoader.cpp]: Fixed it, sorry for that.

File: Libraries/LibAudio/Loader.cpp (hunk 2 above)
Comment (metalvoidzz) [reply Libraries/LibAudio/Loader.cpp:31]: You're right, I've added the option to load from a ByteBuffer.

**Comment by awesomekling** [comment]
👍🏻

**Pull request merged**